### PR TITLE
New covariance smoother

### PR DESCRIPTION
**Pull request by vserret**
add a new way of smoothing the cov matrix using the Wick T123 cov. Fix issue https://github.com/igmhub/picca/issues/600

### CODE REVIEW

**Comment by londumas** [comment]
@vserret, thanks for looking into this.
picca_export already has two functions to giv either the covariance or the correlation function:
```
    parser.add_argument('--cov', type=str, default=None, required=False,
        help='Covariance matrix (if not provided will be calculated by subsampling)')
    parser.add_argument('--cor', type=str, default=None, required=False,
help='Correlation matrix (if not provided will be calculated by subsampling)')
```
So what I would prefer is not to modify `picca_export` and simply add a utils function that from a Wick covariance get a new covariance marix or correlation matrix, into a file and then we give this file to `picca_export` via `--cov` or `--cor`. Does that make sense?

File: py/picca/utils.py [hunk 1]
```diff
@@ -77,6 +78,90 @@ def smooth_cov(da,we,rp,rt,drt=4,drp=4):
...
+        for idrt in range(1,nt):
+            chi = Dcor_red[idrp,idrt]-corrfun(idrp,idrt,L,A)
+            chi2 += chi**2
+        chi2 = chi2*np*nbin   
+        return chi2
+    
+    Lfit=sp.zeros(np)
+    Afit=sp.zeros(np)
+    for idrp in range(np):
+        m = iminuit.Minuit(chisq,L=5.,error_L=0.2,limit_L=(1.,400.),A=1.,error_A=0.2,idrp=idrp,fix_idrp=True,print_level=1,errordef=1.)
```
Comment (londumas) [inline py/picca/utils.py]: split the line, easier to read.

File: py/picca/utils.py [hunk 2]
```diff
@@ -77,6 +78,90 @@ def smooth_cov(da,we,rp,rt,drt=4,drp=4):
...
+        m.migrad()
+        Lfit[idrp]     = m.values['L']
+        Afit[idrp]     = m.values['A']
```
Comment (londumas) [inline py/picca/utils.py]: spacing is off

File: bin/picca_export.py [hunk 3]
```diff
@@ -121,4 +121,4 @@
     h.write([rp,rt,z,da,co,dm,nb],names=['RP','RT','Z','DA','CO','DM','NB'],comment=comment,header=head,extname='COR')
     comment = ['R-parallel model','R-transverse model','Redshift model']
     h.write([dmrp,dmrt,dmz],names=['DMRP','DMRT','DMZ'],comment=comment,extname='DMATTRI')
-    h.close()
```
Comment (londumas) [inline bin/picca_export.py]: h.close() has the wrong indentation. There should be no changes to picca_export.py

**Comment by londumas** [comment]
@vserret, could you fix the trailing spaces? I know you love those :). you can do a "ctrl+f" for " \n".

File: py/picca/utils.py [hunk 4]
```diff
@@ -77,6 +78,88 @@ def smooth_cov(da,we,rp,rt,drt=4,drp=4):
     co_smooth = cor_smooth * sp.sqrt(var*var[:,None])
     return co_smooth
 
+def smooth_cov_wick(da,we,cow,np,nt):
```
Comment (londumas) [inline py/picca/utils.py]: @vserret, I think the function should have as input the path to the covariance matrix and the path to where to save the new covariance. Then the function reads the wick covariance, does it thing and then writes it to a fits file, that can be used in export.py. What do you think?

File: py/picca/utils.py [hunk 5]
```diff
@@ -77,6 +78,88 @@ def smooth_cov(da,we,rp,rt,drt=4,drp=4):
...
+
+    co = cov(da,we)
+
+    nbin = da.shape[1]
+    var = sp.diagonal(co)
+    if sp.any(var==0.):
```
Comment (londumas) [inline py/picca/utils.py]: Let's check both var and varw at the same time.

**Comment by londumas** [comment]
@vserret, you still ahve trailing white space. Did you search " \n"?

**Comment by londumas** [comment]
@vserret, I just removed all trailing spaces. Please do a `git pull` before working on it next time you do. Thanks for the work.

File: py/picca/utils.py (hunk 4 above)
Comment (vserret) [reply py/picca/utils.py]: ok I will do that

**Comment by londumas** [comment]
@vserret, looks good. Is it ready to me merged?

**Comment by vserret** [comment]
yes !

**Comment by londumas** [comment]
@vserret, thank you.

**Pull request merged**